mlrun/wasbfs/fs.py

```diff
@@ -0,0 +1,181 @@
+# Copyright 2022 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from urllib.parse import urlparse
+
+from fsspec import AbstractFileSystem
+
+
+class WasbFS(AbstractFileSystem):
+
+    protocol = "wasb"
+
+    def __init__(
+        self,
+        account_name: str = None,
+        account_key: str = None,
+        connection_string: str = None,
+        credential: str = None,
+        sas_token: str = None,
+        request_session=None,
+        socket_timeout: int = None,
+        blocksize: int = None,
+        client_id: str = None,
+        client_secret: str = None,
+        tenant_id: str = None,
+        anon: bool = True,
+        location_mode: str = None,
+        loop=None,
+        asynchronous: bool = False,
+        default_fill_cache: bool = True,
+        default_cache_type: str = None,
+        **kwargs,
+    ):
+        from adlfs import AzureBlobFileSystem
+
+        self.azure_blob_fs = AzureBlobFileSystem(
+            account_name,
+            account_key,
+            connection_string,
+            credential,
+            sas_token,
+            request_session,
+            socket_timeout,
+            blocksize,
+            client_id,
+            client_secret,
+            tenant_id,
+            anon,
+            location_mode,
+            loop,
+            asynchronous,
+            default_fill_cache,
+            default_cache_type,
+            **kwargs,
+        )
+
+    @staticmethod
+    def _convert_wasb_schema_to_az(url):
+        # convert wasbs schema url to az schema url. used before passing the url to AzureBlobFS.
+        # wasbs://mycontainer@myaccount/path/to/obj is קquivalent to
+        # az://mycontainer/path/to/obj
+        az_path = url
+        parsed_url = urlparse(url)
+        if parsed_url.scheme:
+            if (
+                parsed_url.scheme.lower() == "wasb"
+                or parsed_url.scheme.lower() == "wasbs"
+            ):
+                az_path = (
+                    "az://"
+                    + parsed_url.username
+                    + ("/" if not parsed_url.path.startswith("/") else parsed_url.path)
+                )
+            else:
+                raise ValueError("Operation expects to wasb or wasbs scheme only!")
+        return az_path
```

Comment:
I think this part still missing coverage

Reply:
the test:
def test_load_object_into_dask_dataframe_using_wasbs_url():
cover this one